ucp/_libs/arr.pyi

```diff
@@ -1,7 +1,12 @@
-from typing import Tuple
+# Copyright (c) 2020-2024, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
 
-class Array:
-    def __init__(self, obj: object): ...
+from typing import Generic, Tuple, TypeVar
+
+T = TypeVar("T")
+
+class Array(Generic[T]):
```

Comment:
*nit*: This a slightly weird use of a generic type. Usually one would read a type `Array[T]` as "an array containing objects of type `T`", that is for `x : Array[T]`, then `x[i] : T forall i \in [0, len(x))`. However, here you're using it to mean the wrapped type.

Reply:
This is taken from [verbatim from KvikIO]( https://github.com/rapidsai/kvikio/blob/2053638989525e18157cb5cad0f4414a6662187a/python/kvikio/kvikio/_lib/arr.pyi#L4-L8 ). Appears this was added in PR ( https://github.com/rapidsai/kvikio/pull/214 ), which it looks like you approved 😉

That said, no objection to changing it to something else. Could you please submit a PR to KvikIO with the syntax you prefer? Can then incorporate it here 🙂

Reply:
Apologies for letting this slip. I don't have a brilliant suggestion, so this is fine and I'll have a think.

Reply:
No worries. Maybe we can raise this as an issue on KvikIO and discuss further there?